### PR TITLE
[Storybook][Chat Doc]Document the single-thread usage of the Chat adapter

### PR DESCRIPTION
**Pull request by AmyL219**
# What

Document the single-thread usage of the adapter and composite in Chat
# Why

Document the single-thread usage of the adapter and composite in Chat
# How Tested

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/storybook/stories/Stateful Client/OverviewStateful.stories.mdx [hunk 1]
```diff
@@ -107,7 +107,7 @@ const statefulChatClient = createStatefulChatClient({
 
 // You can then use the `statefulChatClient` in the same way as a `ChatClient` with a slight variation.
 
-const chatThreadClient = await statefulChatClient.createChatThread({});
+const chatThreadClient = await statefulChatClient.createChatThread({ topic: '' });
```
Comment (AmyL219) [inline packages/storybook/stories/Stateful Client/OverviewStateful.stories.mdx:110]: "the error `topic` couldn't be null" is thrown when building the app.

**Review by JoshuaLai — APPROVED** [review]

File: packages/storybook/stories/Adapters/Adapters.stories.mdx [hunk 2]
```diff
@@ -75,6 +77,8 @@ Below are some examples that might be helpful, but additional methods are availa
 
 <Source code={ChatAdapterStateChangesText} />
 
+The ChatAdapter and the ChatComposite are purpose-built to support single chat thread. Only events, such as 'messagesReceived', from the thread that initialized the ChatAdapter, will be triggered. To accommodate multiple threads usage, consider utilizing a more fundamental service, such as the Chat [StatefulClient](./?path=/docs/statefulclient-overview--page), or directly employing the [ChatClient](https://azuresdkdocs.blob.core.windows.net/$web/javascript/azure-communication-chat/1.0.0/classes/chatclient.html). This will allow management of all chat threads that the user has joined.
```
Comment (PorterNan) [inline packages/storybook/stories/Adapters/Adapters.stories.mdx:80]: I feel adding ChatClient (not statefulClient) part is kinda telling customers that "we don't support it well, don't use UI library", but that implantation is actually not bad if you use statefulClient + adapter, and we fully support that when using statefulClient + component

So I guess we can weaken(or remove) the part for directly using ChatClient, and mention developer can use 
1. StatefulClient + Component (like this https://azure.github.io/communication-ui-library/iframe.html?viewMode=docs&id=statefulclient-reacthooks-usepropsfor--page#usepropsfor-chat-app-example)
2. or statefulClient + adapter (pull multi thread info using client then open single chat view using adapter+composite)

File: packages/storybook/stories/Adapters/snippets/ChatAdapterStateChanges.snippet.txt [hunk 3]
```diff
@@ -11,12 +11,16 @@ chatAdapter.removeParticipants()
...
   console.log(event.message.content?.message);
 })
+
 chatAdapter.on('messageSent', (event) => {
   console.log(event.message.content?.message);
 })
 chatAdapter.on('topicChanged', (event) => {
   console.log(event.topic);
 })
+
```
Comment (PorterNan) [inline packages/storybook/stories/Adapters/snippets/ChatAdapterStateChanges.snippet.txt:25]: remove one extra line

File: packages/storybook/stories/Adapters/Adapters.stories.mdx (hunk 2 above)
Comment (AmyL219) [reply packages/storybook/stories/Adapters/Adapters.stories.mdx]: Agree, highlight our library functions would align better with our objectives.

**Review by PorterNan — APPROVED** [review]

**Pull request merged**